android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -22,17 +31,34 @@ import io.flutter.plugin.common.MethodChannel.Result
  * Android implementation of plugin that will let Flutter developers to
  * interact with a Android platform
  * */
-class CustomerIoPlugin : FlutterPlugin, MethodCallHandler {
+class CustomerIoPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
     /// The MethodChannel that will the communication between Flutter and native Android
     ///
     /// This local reference serves to register the plugin with the Flutter Engine and unregister it
     /// when the Flutter Engine is detached from the Activity
     private lateinit var flutterCommunicationChannel: MethodChannel
     private lateinit var context: Context
+    private var activity: Activity? = null
```

Comment:
I think this might cause a memory leak to create a strong reference to an Activity? 

I know that the property is being set and unset based off of `ActivityAware` listener events, but can we rely on that? Or, do we need to make this property a weak reference? 

Is there a solution where we don't need to involve an Activity at all? Avoid this potentially being a problem? 

Assuming there is not a good solution not using an Activity, I am OK with the code proposed if you feel confident in it.